src/plugins/discover/public/application/angular/directives/no_results.tsx

```diff
@@ -0,0 +1,235 @@
+          <EuiText>
+            <strong>
+              <FormattedMessage
+                id="discover.noResults.searchExamples.400to499StatusCodeWithPhpExtensionExampleTitle"
+                defaultMessage="Find status codes 400-499 with the extension php"
+              />
+            </strong>
+          </EuiText>
+        ),
+      },
+      {
+        description: <EuiCode>status:[400 TO 499] AND (extension:php OR extension:html)</EuiCode>,
+        title: (
+          <EuiText>
+            <strong>
+              <FormattedMessage
+                id="discover.noResults.searchExamples.400to499StatusCodeWithPhpOrHtmlExtensionExampleTitle"
+                defaultMessage="Find status codes 400-499 with the extension php or html"
+              />
+            </strong>
+          </EuiText>
+        ),
+      },
+    ];
+
+    luceneQueryMessage = (
+      <Fragment>
+        <EuiSpacer size="xl" />
+
+        <EuiText>
+          <h3>
+            <FormattedMessage
+              id="discover.noResults.searchExamples.refineYourQueryTitle"
+              defaultMessage="Refine your query"
+            />
+          </h3>
+
+          <p>
+            <FormattedMessage
+              id="discover.noResults.searchExamples.howTosearchForWebServerLogsDescription"
+              defaultMessage="The search bar at the top uses Elasticsearch&rsquo;s support for Lucene {queryStringSyntaxLink}.
+                Here are some examples of how you can search for web server logs that have been parsed into a few fields."
+              values={{
+                queryStringSyntaxLink: (
+                  <EuiLink
+                    target="_blank"
+                    href={getServices().docLinks.links.query.luceneQuerySyntax}
+                  >
+                    <FormattedMessage
+                      id="discover.noResults.searchExamples.queryStringSyntaxLinkText"
+                      defaultMessage="Query String syntax"
+                    />
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiText>
+
+        <EuiSpacer size="m" />
+
+        <EuiDescriptionList type="column" listItems={searchExamples} />
+
+        <EuiSpacer size="xl" />
+      </Fragment>
+    );
+  }
+
+  const callOut = !error ? (
+    <EuiFlexItem grow={false} className="dscNoResults">
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="discover.noResults.searchExamples.noResultsMatchSearchCriteriaTitle"
+            defaultMessage="No results match your search criteria"
+          />
+        }
+        color="warning"
+        iconType="help"
+        data-test-subj="discoverNoResults"
+      />
+      {timeFieldMessage}
+      {luceneQueryMessage}
+    </EuiFlexItem>
+  ) : (
+    <EuiFlexItem grow={true} className="dscNoResults">
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="discover.noResults.searchExamples.noResultsBecauseOfError"
```

Comment:
The snapshot in the issue shows different texts. 
These are better!

Reply:
definitely, because the better ones are written by @gchaps, will adapt die screenshot